mne/preprocessing/ica.py

```diff
@@ -2680,8 +2672,18 @@ def read_ica_eeglab(fname):
     ica.pre_whitener_ = np.ones((len(eeg.icachansind), 1))
     ica.pca_mean_ = np.zeros(len(eeg.icachansind))
 
-    ica.unmixing_matrix_ = eeg.icaweights
```

Comment:
I'm afraid this specific change introduced a bug. I don't think the code for `n_components < n_ch` works as expected. I'll have to check, but it is possible that `test_read_ica_eeglab` just validates the "else" part of this if, which corresponds more or less to the behavior of this function before the introduction of this if-else statement. I'll create an issue to describe this bug and submit a corresponding PR.